File: avalon-modules/avalon-dataset/avalon-dataset-repository/src/main/java/io/github/lishangbu/avalon/dataset/repository/PokemonAbilityRepository.java
package io.github.lishangbu.avalon.dataset.repository;

import io.github.lishangbu.avalon.dataset.entity.PokemonAbility;
import io.github.lishangbu.avalon.dataset.entity.PokemonAbility.PokemonAbilityId;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

/// 宝可梦特性数据访问接口
///
/// 提供宝可梦特性数据的CRUD操作
///
/// @author lishangbu
/// @since 2025/08/20
@Repository
public interface PokemonAbilityRepository extends JpaRepository<PokemonAbility, PokemonAbilityId> {}

File: avalon-modules/avalon-dataset/avalon-dataset-repository/src/main/kotlin/io/github/lishangbu/avalon/dataset/repository/PokemonAbilityRepository.kt
package io.github.lishangbu.avalon.dataset.repository

import io.github.lishangbu.avalon.dataset.entity.PokemonAbility
import io.github.lishangbu.avalon.dataset.entity.PokemonAbilityId
import org.babyfish.jimmer.spring.repository.KRepository
import org.springframework.stereotype.Repository

/**
 * 宝可梦特性仓储接口
 *
 * 定义宝可梦特性数据的查询与持久化操作
 *
 * @author lishangbu
 * @since 2025/08/20
 */
@Repository
interface PokemonAbilityRepository : KRepository<PokemonAbility, PokemonAbilityId>
